django/core/management/commands/shell.py

```diff
@@ -127,11 +130,17 @@ def handle(self, **options):
             exec(sys.stdin.read())
             return
 
-        available_shells = [options['interface']] if options['interface'] else self.shells
-
-        for shell in available_shells:
+        if options['interface']:
+            shell = options['interface']
             try:
                 return getattr(self, shell)(options)
-            except ImportError:
-                pass
-        raise ImportError("Couldn't load any of the specified interfaces.")
+            except NoSuchShell:
+                raise CommandError("Couldn't load {} interface.".format(shell))
+
+        else:
+            for shell in self.shells:
+                try:
+                    return getattr(self, shell)(options)
+                except NoSuchShell:
+                    pass
+            raise CommandError("Couldn't load any interface.")
```

Comment:
I think it's impossible to reach this since the `python` shell doesn't raise `NoSuchShell`.